OpenRA.Mods.Common/Scripting/Properties/MissionObjectiveProperties.cs

```diff
@@ -19,22 +19,31 @@ namespace OpenRA.Mods.Common.Scripting
 	[ScriptPropertyGroup("MissionObjectives")]
 	public class MissionObjectiveProperties : ScriptPlayerProperties, Requires<MissionObjectivesInfo>
 	{
+		[TranslationReference]
+		const string Primary = "primary";
+
+		[TranslationReference]
+		const string Secondary = "secondary";
+
+		readonly ScriptContext context;
 		readonly MissionObjectives mo;
 		readonly bool shortGame;
 
 		public MissionObjectiveProperties(ScriptContext context, Player player)
 			: base(context, player)
 		{
+			this.context = context;
 			mo = player.PlayerActor.Trait<MissionObjectives>();
 			shortGame = player.World.WorldActor.Trait<MapOptions>().ShortGame;
 		}
 
 		[ScriptActorPropertyActivity]
 		[Desc("Add a mission objective for this player. The function returns the " +
 			"ID of the newly created objective, so that it can be referred to later.")]
-		public int AddObjective(string description, string type = "Primary", bool required = true)
+		public int AddObjective(string description, string type = Primary, bool required = true)
```

Comment:
Note: This feels very weird. The other parameter(s) require you to pass an already translated string as well. Ideally this would be `string type = context.World.Map.Translate(Primary)` but I can see why you couldn't do that. Maybe the best solution is to remove the default parameter here.

Reply:
I don't want to change the API when I don't have to.

Reply:
But this is changing the API already? 🤔 Existing `type`s will suddenly be translated, which might work out because there is no translation and it falls back to the key, but it also might not, and probably generates messages in the debug log.

Reply:
That is not changing an API, that's changing the API's behaviour, which is fine. The method signature itself stays the same, which is the important bit.
Here it looks a bit odd, but I assume introducing a breaking change like that to the Lua API would be pretty bad.